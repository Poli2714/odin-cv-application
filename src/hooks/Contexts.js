import { createContext } from 'react';

const IndexContext = createContext({
  activeIndex: null,
  setActiveIndex: () => {},
});

const KeyContext = createContext({
  activeKey: {
    eduActiveKey: null,
    expActiveKey: null,
  },
  setActiveKey: () => {},
});

const EduErrMsgContext = createContext({
  showEduErrMessage: false,
  setShowEduErrMessage: () => {},
});

const ExpErrMsgContext = createContext({
  showExpErrMessage: false,
  setShowExpErrMessage: () => {},
});

const PersonalInfoContext = createContext({
  personalInfo: {
    profilePicture: '',
    fullName: '',
    currentJobTitle: '',
    email: '',
    mobile: '',
    city: '',
  },
  setPersonalInfo: () => {},
});

const SummaryContext = createContext({
  summary: '',
  setSummary: () => {},
});

const ExperienceContext = createContext({
  experiences: {
    id: {
      jobTitle: '',
      employer: '',
      jobStartDate: '',
      jobEndDate: '',
      jobDescription: '',
    },
  },
  setExperiences: () => {},
});

const EducationContext = createContext({
  educations: {
    id: {
      degree: '',
      university: '',
      eduStartDate: '',
      eduEndDate: '',
      eduInfo: '',
    },
  },
  setEducations: () => {},
});

export {
  EducationContext,
  EduErrMsgContext,
  ExperienceContext,
  ExpErrMsgContext,
  IndexContext,
  KeyContext,
  PersonalInfoContext,
  SummaryContext,
};
