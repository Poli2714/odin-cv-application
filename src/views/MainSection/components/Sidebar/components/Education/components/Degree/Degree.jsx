import { useContext } from 'react';
import PropTypes from 'prop-types';

import { InputWrapper } from 'src/components/layout';
import { Label, TextInput } from 'src/components/forms';
import { EducationContext } from 'src/hooks/Contexts';

function Degree({ id }) {
  const { educations, setEducations } = useContext(EducationContext);
  const { degree } = educations[id];

  const handleChangeDegree = e =>
    setEducations({
      ...educations,
      [id]: { ...educations[id], degree: e.target.value },
    });

  return (
    <InputWrapper>
      <Label labelName="Degree" labelFor="degree" />
      <TextInput
        autofocus={true}
        name="degree"
        placeholder="e.g. M.Sc. in Computer Science"
        value={degree}
        onChange={handleChangeDegree}
      />
    </InputWrapper>
  );
}

Degree.propTypes = {
  id: PropTypes.string,
};

export default Degree;
