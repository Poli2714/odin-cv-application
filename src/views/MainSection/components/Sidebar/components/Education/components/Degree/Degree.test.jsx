import { expect, test, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useState } from 'react';

import Degree from './Degree';
import { EducationContext } from 'src/hooks/Contexts';

vi.mock('src/components/layout', () => ({
  InputWrapper: vi.fn(({ children }) => (
    <div data-testid="input-wrapper">{children}</div>
  )),
}));

vi.mock('src/components/forms', () => ({
  Label: vi.fn(() => <label htmlFor="degree">Mock label</label>),
  TextInput: vi.fn(({ onChange, value }) => (
    <input id="degree" type="text" onChange={onChange} value={value} />
  )),
}));

function MockParentComponent({ id }) {
  const [educations, setEducations] = useState({
    fakeID: { degree: '' },
    fakeID2: { degree: 'test' },
  });

  return (
    <EducationContext.Provider value={{ educations, setEducations }}>
      <Degree id={id} />
    </EducationContext.Provider>
  );
}

test('renders Degree', () => {
  render(<MockParentComponent id="fakeID" />);
  expect(screen.getByTestId('input-wrapper')).toMatchSnapshot();
  expect(screen.getByLabelText('Mock label')).toHaveValue('');
});

test('renders Degree with different initial id', () => {
  render(<MockParentComponent id="fakeID2" />);

  expect(screen.getByLabelText('Mock label')).toHaveValue('test');
});

test('updates degree value as a user types', async () => {
  const user = userEvent.setup();

  render(<MockParentComponent id="fakeID" />);
  const input = screen.getByLabelText('Mock label');

  await user.type(input, 'High Magician');
  expect(input).toHaveValue('High Magician');
});
