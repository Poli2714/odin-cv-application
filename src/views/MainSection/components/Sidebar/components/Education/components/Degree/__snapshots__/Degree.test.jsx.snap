// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Degree 1`] = `
<div
  data-testid="input-wrapper"
>
  <label
    for="degree"
  >
    Mock label
  </label>
  <input
    id="degree"
    type="text"
    value=""
  />
</div>
`;
